src/robot/output/listeners.py

```diff
@@ -15,298 +15,591 @@
+        attrs = self._body_item_v2_attributes(data, result, is_end=True)
+        attrs['values'] = list(result.values)
+        self._end_return(result._log_name, attrs)
 
-    def start_break(self, data, result):
-        self.listener.start_keyword(ModelCombiner(data, result))
+    def start_continue(self, data: 'running.Continue', result: 'result.Continue'):
+        self._start_continue(result._log_name, self._body_item_v2_attributes(data, result))
 
-    def end_break(self, data, result):
-        self.listener.end_keyword(ModelCombiner(data, result))
+    def end_continue(self, data: 'running.Continue', result: 'result.Continue'):
+        self._end_continue(result._log_name, self._body_item_v2_attributes(data, result, is_end=True))
 
-    def start_continue(self, data, result):
-        self.listener.start_keyword(ModelCombiner(data, result))
+    def start_break(self, data: 'running.Break', result: 'result.Break'):
+        self._start_break(result._log_name, self._body_item_v2_attributes(data, result))
 
-    def end_continue(self, data, result):
-        self.listener.end_keyword(ModelCombiner(data, result))
+    def end_break(self, data: 'running.Break', result: 'result.Break'):
+        self._end_break(result._log_name, self._body_item_v2_attributes(data, result, is_end=True))
 
-    def start_return(self, data, result):
-        self.listener.start_keyword(ModelCombiner(data, result))
+    def start_error(self, data: 'running.Error', result: 'result.Error'):
+        self._start_error(result._log_name, self._body_item_v2_attributes(data, result))
 
-    def end_return(self, data, result):
-        self.listener.end_keyword(ModelCombiner(data, result))
+    def end_error(self, data: 'running.Error', result: 'result.Error'):
+        self._end_error(result._log_name, self._body_item_v2_attributes(data, result, is_end=True))
 
-    def start_error(self, data, result):
-        self.listener.start_keyword(ModelCombiner(data, result))
+    def start_var(self, data: 'running.Var', result: 'result.Var'):
+        self._start_var(result._log_name, self._body_item_v2_attributes(data, result))
 
-    def end_error(self, data, result):
-        self.listener.end_keyword(ModelCombiner(data, result))
+    def end_var(self, data: 'running.Var', result: 'result.Var'):
+        self._end_var(result._log_name, self._body_item_v2_attributes(data, result, is_end=True))
 
-    def imported(self, import_type: str, name: str, attrs):
-        self.listener.imported(import_type, name, attrs)
-
-    def output_file(self, type_: str, path: str):
-        self.listener.output_file(type_, path)
+    def log_message(self, message: 'model.Message'):
+        self._log_message(self._message_attributes(message))
 
-    def close(self):
-        self.listener.close()
+    def message(self, message: 'model.Message'):
+        self._message(self._message_attributes(message))
+
+    def _suite_v2_attributes(self, data, result, is_end=False):
```

Comment:
`_v2` in the name is unnecessary in this context. Same thing with other similar methods.